### ISSUE TITLE
Upload to custom folder and save to Media table

### ISSUE THREAD

**Issue by bkhim** [cloudinary-labs/cloudinary-laravel#27]
Can there be a helper to upload images to custom cloudinary folder `$uploadedFile = $file->storeOnCloudinary(Auth::user()->name);` and at the same time handle the saving to the `media` table.
Currently I have to use that function and still write more code to save to a media table.
More like `$uploadedFile = $file->attachOnCloudinary(Auth::user()->name);`

Thank you.

**Comment by SlimGee** [comment]
Are you attaching these files to a model or you just want to upload them

**Comment by hessel-bb** [comment]
im facing kinda a similar issue. i want to save the image in a folder called 'grrp' on cloudinary and save it in my media table as well. But when i can't figure out how to do this. I thought that i could save it to the grrp folder and then on the next line attach it to the model but that uploads the image two times.  This is the work around that i thought might work:

```
        $file = $request->file('thumbnail');
        $upload = cloudinary()->upload($file->getRealPath(), [
                'folder' => 'grrp'
         ])->getSecurePath();
         $user->profile->updateMedia($file); 
```
So then i thought i do the following:
  ```
 $file = $request->file('thumbnail');
   $user->profile->updateMedia($file->getRealPath(), [ 'folder' => 'grrp'])->getSecurePath();
```
But that doesn't seem to work. Is there another way to do this.

TL;DR: I want to store the image on cloudinary in a folder called 'grrp' and save it in my mysql database.

**Comment by bkhim** [comment]
I managed to perform this in a way. Sorry I never updated this comment.
Suppose you have a separate table to store images and referenced by a foreign key from the parent table. You can create a relationship in the model classes that allow you to update the images table directly on uploading. Something like this:

```
$userAvatar       = $request->file('user_avatar');

$details               = new UserDetails();

$details->image = cloudinary()->upload($originalImage->getRealPath(),
                array("folder" => 'courses',))->getSecurePath()

User::find($id)->details()->save($details);
```

This is how I shortened my code. Initially it was a long one.
